portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.test.js

```diff
@@ -0,0 +1,136 @@
+const httpMocks = require('node-mocks-http');
+const { MOCK_PORTAL_SESSION_USER } = require('../../../test-mocks/mock-portal-session-user');
+const { postUtilisationReportUpload } = require('.');
+const { getUploadErrors } = require('./utilisation-report-upload-errors');
+const { getDueReportDates } = require('./utilisation-report-status');
+const { validateCsvData } = require('./utilisation-report-validator');
+const { extractCsvData } = require('../../../utils/csv-utils');
+
+jest.mock('./utilisation-report-upload-errors');
+jest.mock('./utilisation-report-status');
+jest.mock('../../../utils/csv-utils');
+jest.mock('./utilisation-report-validator');
+
+describe('controllers/utilisation-report-service/utilisation-report-upload', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('postUtilisationReportUpload', () => {
+    it("renders the 'utilisation-report-upload' page if getUploadErrors returns errors", async () => {
+      // Arrange
+      const { res, req } = httpMocks.createMocks({
+        session: { userToken: 'user-token', user: MOCK_PORTAL_SESSION_USER },
+      });
+
+      jest.mocked(getUploadErrors)
+        .mockReturnValueOnce({
+          uploadErrorSummary: {
+            text: 'Error',
+            href: '#utilisation-report-file-upload',
+          },
+          uploadValidationError: {
+            text: 'Error',
+          },
+        });
+
+      jest.mocked(getDueReportDates).mockReturnValueOnce([]);
+
+      // Act
+      await postUtilisationReportUpload(req, res);
+
+      // Assert
+      /* eslint-disable no-underscore-dangle */
```

Comment:
Seeing as this is only one line, we can use `// eslint-disable-next-line no-underscore-dangle`